registry/storage/driver/azure/azure.go

```diff
@@ -282,13 +282,31 @@ func (d *driver) Move(ctx context.Context, sourcePath string, destPath string) e
 		return err
 	}
 	destBlobRef := d.client.NewBlockBlobClient(d.blobName(destPath))
-	_, err = destBlobRef.CopyFromURL(ctx, sourceBlobURL, nil)
+	resp, err := destBlobRef.StartCopyFromURL(ctx, sourceBlobURL, nil)
 	if err != nil {
 		if is404(err) {
 			return storagedriver.PathNotFoundError{Path: sourcePath}
 		}
 		return err
 	}
+	copyStatus := *resp.CopyStatus
+	for copyStatus == blob.CopyStatusTypePending {
+		props, err := destBlobRef.GetProperties(ctx, nil)
+		if err != nil {
+			return err
+		}
+		copyStatus = *props.CopyStatus
+		if copyStatus == blob.CopyStatusTypeAborted || copyStatus == blob.CopyStatusTypeFailed {
+			if props.CopyStatusDescription != nil {
+				return fmt.Errorf("failed to move blob: %s", *props.CopyStatusDescription)
+			}
+			return fmt.Errorf("failed to move blob with copy id %s", *props.CopyID)
+		}
+
+		if copyStatus == blob.CopyStatusTypePending {
+			time.Sleep(100 * time.Millisecond)
```

Comment:
We should make this a constant or a configurable param with this being the default. Folks in different locations might face different latencies, etc.

Reply:
put into a constant named `waitBetweenCopyStatusPolling`. the name is a bit questionable but I'm out of ideas.

Reply:
Good suggestion - done.